node/cli/src/service.rs

```diff
@@ -162,6 +172,13 @@ macro_rules! new_full {
 			let babe = babe::start_babe(babe_config)?;
 			let select = babe.select(service.on_exit()).then(|_| Ok(()));
 			service.spawn_task(Box::new(select));
+
+			let authority_discovery = authority_discovery::AuthorityDiscovery::new(
+				service.client(),
+				service.network(),
+				dht_event_rx,
+			);
+			let _ = service.spawn_task(Box::new(authority_discovery));
```

Comment:
I have copied this from other `service.spawn_task()` calls, e.g. see https://github.com/paritytech/substrate/blob/master/core/service/src/lib.rs#L252. I guess the sole purpose to do so is to prevent the *Unused result* warning. Do you still want it removed?

Reply:
The code you linked to reads `let _ = to_spawn_tx_.unbounded_send(future);` which is something else entirely. The signature of `spawn_task` is:

```rust
fn spawn_task(&self, task: impl Future<Item = (), Error = ()> + Send + 'static);
```

Since it returns unit, there is no need to bind it to `_`.

Reply:
Right, sorry about that. I implemented it inside `core/service/src/lib.rs` before.